nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -326,6 +329,27 @@ public void process(InputStream in) throws IOException {
         return splitFlowFiles;
     }
 
+    /**
+     * Will merge the two files. This operation has different semantics then
+     * sessioin.merge. Also, it always expects 2 and no more then 2 flow files,
+     * so vararg signature primarily for simplification.
+     */
+    private FlowFile merge(ProcessSession session, FlowFile... flowFiles) {
+        FlowFile mergedFlowFile = session.create();
```

Comment:
Koji, as you can see, there are no source/target files here. Just an array of FlowFiles that need to be concatenated in the order they were provided. It is a simple internal utility operation which requires no provenance. 
The fact that I am storing header information in what essentially is a temporary FlowFile, does not mean I am deriving A from B.